File: yo-cordapp/workflows-java/src/test/java/net/corda/examples/yo/FlowTests.java
package net.corda.examples.yo;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.transactions.SignedTransaction;
import net.corda.examples.yo.flows.YoFlow;
import net.corda.examples.yo.flows.YoFlowResponder;
import net.corda.testing.node.MockNetwork;
import net.corda.testing.node.MockNetworkParameters;
import net.corda.testing.node.StartedMockNode;
import net.corda.testing.node.TestCordapp;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters().withCordappsForAllNodes(ImmutableList.of(
            TestCordapp.findCordapp("net.corda.examples.yo.contracts"),
            TestCordapp.findCordapp("net.corda.examples.yo.flows")
    )));
    private final StartedMockNode a = network.createNode();
    private final StartedMockNode b = network.createNode();

    public FlowTests() {
        ImmutableList.of(a, b).forEach(it -> {
            it.registerInitiatedFlow(YoFlowResponder.class);
        });
    }

    @Before
    public void setup() {
        network.runNetwork();
    }

    @After
    public void tearDown() {
        network.stopNodes();
    }

    //The yo flow should not have any input
    //This test will check if the input list is empty
    @Test
    public void dummyTest() throws ExecutionException, InterruptedException {
        CordaFuture<SignedTransaction> future = a.startFlow(new YoFlow(b.getInfo().getLegalIdentities().get(0)));
        network.runNetwork();
        SignedTransaction ptx = future.get();
        assert(ptx.getTx().getInputs().isEmpty());
    }
}

File: yo-cordapp/workflows-kotlin/src/test/kotlin/net/corda/examples/yo/FlowTests.kt
package net.corda.examples.yo

import net.corda.examples.yo.flows.YoFlow
import net.corda.examples.yo.flows.YoFlowResponder
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkParameters
import net.corda.testing.node.TestCordapp
import org.junit.After
import org.junit.Before
import org.junit.Test

class FlowTests {
    private val network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
        TestCordapp.findCordapp("net.corda.examples.yo.contracts"),
        TestCordapp.findCordapp("net.corda.examples.yo.flows")
    )))
    private val a = network.createNode()
    private val b = network.createNode()

    init {
        listOf(a, b).forEach {
            it.registerInitiatedFlow(YoFlowResponder::class.java)
        }
    }

    @Before
    fun setup() = network.runNetwork()

    @After
    fun tearDown() = network.stopNodes()

    //The yo flow should not have any input
    //This test will check if the input list is empty
    @Test
    fun `dummy test`() {
        val future = a.startFlow(YoFlow(b.info.legalIdentities.first()))
        network.runNetwork()
        val ptx = future.get()
        assert(ptx.tx.inputs.isEmpty())
    }
}